src/traversals/eulerian.jl

```diff
@@ -0,0 +1,93 @@
+# Adapted from SimpleGraphs.jl [Copyright (c) 2014, Ed Scheinerman]:
+#    https://github.com/scheinerman/SimpleGraphs.jl/blob/master/src/simple_euler.jl
+# Reproduced under the MIT Expat License.
+
+"""
+    eulerian(g::AbstractSimpleGraph{T}[, u::T]) --> T[]
+
+Returns a [Eulerian trail or cycle](https://en.wikipedia.org/wiki/Eulerian_path) through an
```

Comment:
Is it a trail or a cycle?

Reply:
Afaict, [the nomenclature](https://en.wikipedia.org/wiki/Eulerian_path#Definition) is that it is a cycle if the start and end vertices are identical - and otherwise, it's a trail.

Reply:
What I mean is, how do I know which one will be returned?

Reply:
Okay, added some explanations to the docstring of what is meant.